web/src/search_pill.ts

```diff
@@ -109,6 +111,90 @@ function on_pill_exit(
     $user_pill.remove();
 }
 
+function search_user_pill_data_from_term(term: NarrowTerm): SearchUserPill {
+    const emails = term.operand.split(",");
+    const users = emails.map((email) => {
+        const person = people.get_by_email(email);
+        assert(person !== undefined);
+        return person;
+    });
+    return search_user_pill_data(users, term.operator, term.negated ?? false);
+}
+
+// TODO: We're calculating `description_html` every time, even though
+// we only show it (in `generate_pills_html`) for lines with only one
+// pill. We can probably simplify things by separating out a function
+// that generates `description_html` from the information in a single
+// search pill, and remove `description_html` from the `Suggestion` type.
+export function generate_pills_html(suggestion: Suggestion): string {
+    const search_terms = Filter.parse(suggestion.search_string);
+
+    const pills: SearchPill[] = search_terms.map((term) => {
+        if (user_pill_operators.has(term.operator) && term.operand !== "") {
+            return search_user_pill_data_from_term(term);
+        }
+        return {
+            type: "search_non_user",
+            operator: term.operator,
+            operand: term.operand,
+            negated: term.negated,
+        };
+    });
+
+    const pill_render_data = pills.map((item, index) => {
+        if (item.type === "search_user") {
+            return item;
+        }
+        if (item.operator === "topic" && item.operand === "") {
+            return {
+                ...item,
+                is_empty_string_topic: true,
+                sign: item.negated ? "-" : "",
+                topic_display_name: util.get_final_topic_display_name(""),
+            };
+        }
+        if (item.operator === "search") {
+            let description_html = search_term_description_html(item);
+            // We capitalize the beginning of the suggestion line if it's text (not
+            // pills), which is only relevant for suggestions with search operators.
+            if (index === 0) {
+                const capitalized_first_letter = description_html.charAt(0).toUpperCase();
+                description_html = capitalized_first_letter + description_html.slice(1);
+            }
+            return {
+                ...item,
+                description_html,
+            };
+        }
+        return {
+            ...item,
+            display_value: get_search_string_from_item(item),
+        };
+    });
+
+    // We show help text (description_html) only when there's just a
+    // single pill on that suggstion line.
```

Comment:
`suggestion`. Also a few words about why this is what we do would be helpful. Maybe

> When there's a single pill on a suggestion line, we have space to provide help text (description_html) explaining what the operator does.